File: LeetCode/28.Implement_strStr/main.cpp
#include "solution.h"

// Unit Test

int main(){
    Solution s;
    string h1 = "hello", n1 = "ll";
    string h2 = "aaaaa", n2 = "bba";
    string h3 = "", n3 = "";
    if (s.strStr(h1, n1) == 2) { cout << "Pass" << endl;}
    else { cout << "Failed - h1 n1" << endl;}
    if (s.strStr(h2, n2) == -1) { cout << "Pass" << endl;}
    else { cout << "Failed - h2 n2" << endl;}
    if (s.strStr(h3, n3) == 0) { cout << "Pass" << endl;}
    else { cout << "Failed - h3 n3" << endl;}
    return 0;
}

File: LeetCode/28.Implement_strStr/main.py
#!/usr/bin/python

from solution import Solution

# Unit Test
haystack1 = "hello"
needle1 = "ll"
haystack2 = "aaaaa"
needle2 = "bba"
haystack3 = ""
needle3 = ""

s1 = Solution(haystack1, needle1)
s2 = Solution(haystack2, needle2)
s3 = Solution(haystack3, needle3)

if (s1.strStr()==2):
    print("Pass")
else:
    print("Test 1 Failed")
if (s2.strStr()==-1):
    print("Pass")
else:
    print("Test 2 Failed")
if (s3.strStr()==0):
    print("Pass")
else:
    print("Test 3 Failed")